### DISCUSSION TITLE
Can I configure slint fps or traceing slint backend log, just like QT's QT_LOGGING_RULES

### DISCUSSION THREAD

**Discussion by fluolite** [slint-ui/slint#6349] [General] [answered]
I run slint-ui in different devices, but the experience varies greatly!

```
Device-1
system:  Linux-x86_64（Intel）,  x11 desktop system
slint backend: "backend-winit"
slint renderer: "renderer-skia"
render result:     60fps
 
Device-2
system:  Linux-aarch64（rockchip RK3568）,  has egl, gles2, gbm, drm
slint backend: "backend-linuxkms-noseat"
slint renderer: "renderer-skia" or "renderer-software"
render result:     1-6 fps
But QT eglfs render result: 60fps

```

Did I configure slint something wrong or How can I detect the problem? 
Can I traceing slint backend log, just like QT's QT_LOGGING_RULES ?

![Screenshot_select-area_20240927184012](https://github.com/user-attachments/assets/f3c493a4-c4d1-4038-bbbe-54c31dd158c2)
![Screenshot_select-area_20240927184937](https://github.com/user-attachments/assets/cbef6983-051d-4ec9-a77e-b620d0075c4d)

**Comment by tronical** [comment]
It's hard to say what might be causing that. How does CPU usage compare? How does your Slint scene look like, and how does it compare to the same scene in QML?

**Reply by fluolite** [reply]
CPU usage:
![Screenshot_select-area_20240929095524](https://github.com/user-attachments/assets/5d6bb539-a2fb-40be-aed9-398167e345b4)

It is the same result for a single slint helloworld exsample.
It's easy to run 60fs with complex scene with QML（backend: EGLFS）.

This is the board SoC info.
[https://www.rock-chips.com/uploads/pdf/2022.8.26/191/RK3568%20Brief%20Datasheet.pdf](https://www.rock-chips.com/uploads/pdf/2022.8.26/191/RK3568%20Brief%20Datasheet.pdf)

**Reply by fluolite** [reply]
How can I configure slint-ui backend and renderer, just like QT's EGLFS?

**Reply by tronical** [reply]
That's a relatively low CPU usage, unless... the 25% mean that one of the four cores is running at 100%. Do you know how this is displayed in top for you? Either way, it's odd.

So just to double check:

You're showing a scene like [this](https://github.com/slint-ui/slint-rust-template/blob/main/ui/app-window.slint)?
What value exactly do you set `SLINT_DEBUG_PERFORMANCE` to in order to measure the fps?

**Reply by fluolite** [reply, marked as answer]
I made a mistake, slint render as software renderer because EglDisplay create failed

[[glutin ](https://github.com/rust-windowing/glutin/issues/1708)](https://github.com/rust-windowing/glutin/issues/1708)

**Reply by tronical** [reply]
Good find. Let's continue the discussion there.
